core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java

```diff
@@ -84,6 +101,9 @@ public PrimitiveQueryType(String name, String parentname, int blockIndex,
       DimensionRawColumnChunk[] rawColumnChunks, int rowNumber,
       int pageNumber, DataOutputStream dataOutputStream) throws IOException {
     byte[] currentVal = copyBlockDataChunk(rawColumnChunks, rowNumber, pageNumber);
+    if (!this.isDictionary) {
+      dataOutputStream.writeInt(currentVal.length);
```

Comment:
Just writing length? where are you writing data?

Reply:
Below line 105 we are writing the data. Data writing is without any condition.